tests/test_declaration.py

```diff
@@ -120,6 +120,13 @@ def test_create_and_copy(StyleClass):
     assert dup.implicit == VALUE3
 
 
+@pytest.mark.parametrize("StyleClass", [Style, DeprecatedStyle])
+def test_deprecated_copy(StyleClass):
+    style = StyleClass(explicit_const=VALUE2)
+    with pytest.warns(DeprecationWarning):
+        style.copy(applicator=object())
```

Comment:
I'd be inclined to make the applicator a Mock, and assert that the style is being applied as a result of this copy behavior.

Reply:
Hm. Should assigning an applicator to a *style* automatically make it reapply? I see how that could make sense, in the property setter on style. Currently I'm handling that logic from node, and style only reapplies when it's explicitly told to.

Reply:
Hrm... that highlights an interesting discrepancy / API gap. At least in Toga's usage, Applicator has a reference to the Node it will be applied to, and the node has a reference to it's applicator, but there's no requirement that the applicator's node and the node's applicator actually match. You could set an applicator on Node 1 that actually applies on Node 2.

I'm not sure what the fix for this is. Using a clean sheet of paper, there's an argument to be made that the applicator should be stateless, and be *given* the node it should apply to (or, at least, have the node *set* as part of being assigned to a node)... but getting from here to there could be messy.

Reply:
Also: dang, you must get up early.

Reply:
> At first glance, I don't see how we'd _break_ anything by adding `applicator.node = self` in node's applicator setter. Toga doesn't currently even call it that; the reference is named `widget`. A future update of Toga would be able to dispense with the manual assignment.

I guess that's true - we don't *break* anything by setting `applicator.node`; but when we fix Toga to address applicator ordering and `_impl` issue that spawned this ticket, we can modify Applicator to *not* accept widget, and apply the styles on Node.